packages/io/__tests__/controller.ts

```diff
@@ -0,0 +1,52 @@
+import { describe, it, expect } from '@jest/globals'
+import { CoR } from '../src/cor'
+import { BaseController, Get } from '../src'
+import type { Context } from '../src/types'
+
+class AppController extends BaseController {
+  @Get()
+  async hello(ctx: Context) {
+    ctx.ok('hello root')
+  }
+
+  @Get('/parent')
+  async parent(ctx: Context) {
+    ctx.ok('hello parent')
+  }
+
+  @Get('/parent/children')
+  async children(ctx: Context) {
+    ctx.ok('hello children')
+  }
+
+  @Get('/:username')
+  async username(ctx: Context<{ payload: { params: { username?: string } } }>) {
```

Comment:
How about using param decorators to inject the parameters.

```
class AppController {
  @Get('/:username')
  async username(@Query('age') age: number, @Param('username') username: string) {}
}
```

Reply:
👌